docs/3.registering-abilities.md

```diff
@@ -0,0 +1,276 @@
+                    'description' => 'Number of orders in period'
+                ),
+                'total_revenue' => array(
+                    'type' => 'number',
+                    'description' => 'Total revenue in period'
+                )
+            )
+        ),
+        'execute_callback' => function( $input ) {
+            $period = $input['period'] ?? 'month';
+            
+            // Implementation would calculate stats based on period
+            return array(
+                'total_orders' => 42,
+                'total_revenue' => 1250.50
+            );
+        },
+        'permission_callback' => function() {
+            return current_user_can( 'manage_woocommerce' );
+        },
+        'meta' => array(
+            'requires_plugin' => 'woocommerce',
+            'category' => 'ecommerce'
+        )
+    ));
+}
+```
+
+**4. Registering an Ability That May Fail**
+
+```php
+add_action( 'abilities_api_init', 'my_plugin_register_send_email_ability' );
+function my_plugin_register_send_email_ability() {
+    wp_register_ability( 'my-plugin/send-email', array(
+        'label' => __( 'Send Email', 'my-plugin' ),
+        'description' => __( 'Sends an email to the specified recipient using WordPress mail functions.', 'my-plugin' ),
+        'input_schema' => array(
+            'type' => 'object',
+            'properties' => array(
+                'to' => array(
+                    'type' => 'string',
+                    'format' => 'email',
+                    'description' => 'Recipient email address'
+                ),
+                'subject' => array(
+                    'type' => 'string',
+                    'minLength' => 1,
+                    'description' => 'Email subject'
+                ),
+                'message' => array(
+                    'type' => 'string',
+                    'minLength' => 1,
+                    'description' => 'Email message body'
+                )
+            ),
+            'required' => array( 'to', 'subject', 'message' ),
+            'additionalProperties' => false
+        ),
+        'output_schema' => array(
+            'type' => 'object',
+            'properties' => array(
+                'sent' => array(
+                    'type' => 'boolean',
+                    'description' => 'Whether the email was successfully sent'
+                )
+            )
+        ),
+        'execute_callback' => function( $input ) {
+            $sent = wp_mail(
+                $input['to'],
+                $input['subject'],
+                $input['message']
+            );
+            
+            if ( ! $sent ) {
+                throw new Exception( 'Failed to send email' );
+            }
+            
+            return array( 'sent' => true );
+        },
+        'permission_callback' => function() {
+            return current_user_can( 'publish_posts' );
+        }
+    ));
+}
+```
+
+**Usage Examples**
```

Comment:
By the way, the next document is about usage, so proably this section should get moved there.